src/components/swap/SwapLineItem.tsx

```diff
@@ -0,0 +1,250 @@
+import { Plural, t, Trans } from '@lingui/macro'
+import { Currency, CurrencyAmount, Percent, TradeType } from '@uniswap/sdk-core'
+import { LoadingRow } from 'components/Loader/styled'
+import RouterLabel from 'components/RouterLabel'
+import { RowBetween } from 'components/Row'
+import { MouseoverTooltip, TooltipSize } from 'components/Tooltip'
+import { SUPPORTED_GAS_ESTIMATE_CHAIN_IDS } from 'constants/chains'
+import useHoverProps from 'hooks/useHoverProps'
+import { useIsMobile } from 'nft/hooks'
+import React, { PropsWithChildren, useEffect, useState } from 'react'
+import { InterfaceTrade } from 'state/routing/types'
+import { getTransactionCount, isPreviewTrade, isUniswapXTrade } from 'state/routing/utils'
+import styled, { DefaultTheme } from 'styled-components'
+import { ExternalLink, ThemedText } from 'theme/components'
+import { NumberType, useFormatter } from 'utils/formatNumbers'
+import { getPriceImpactColor } from 'utils/prices'
+
+import { GasBreakdownTooltip, UniswapXDescription } from './GasBreakdownTooltip'
+import SwapRoute from './SwapRoute'
+
+export enum SwapLineItemTypes {
+  EXCHANGE_RATE,
+  NETWORK_FEE,
+  INPUT_TOKEN_FEE_ON_TRANSFER,
+  OUTPUT_TOKEN_FEE_ON_TRANSFER,
+  PRICE_IMPACT,
+  MAXIMUM_INPUT,
+  MINIMUM_OUTPUT,
+  EXPECTED_OUTPUT,
+  ROUTING_INFO,
+}
+
+const DetailRowValue = styled(ThemedText.BodySmall)`
+  text-align: right;
+  overflow-wrap: break-word;
+`
+const LabelText = styled(ThemedText.BodySmall)<{ hasTooltip?: boolean }>`
+  cursor: ${({ hasTooltip }) => (hasTooltip ? 'help' : 'auto')};
+  color: ${({ theme }) => theme.neutral2};
+`
+const ColorWrapper = styled.span<{ textColor?: keyof DefaultTheme }>`
+  ${({ textColor, theme }) => textColor && `color: ${theme[textColor]};`}
+`
+
+function FOTTooltipContent() {
+  return (
+    <>
+      <Trans>
+        Some tokens take a fee when they are bought or sold, which is set by the token issuer. Uniswap does not receive
+        any of these fees.
+      </Trans>{' '}
+      <ExternalLink href="https://support.uniswap.org/hc/en-us/articles/18673568523789-What-is-a-token-fee-">
+        Learn more
+      </ExternalLink>
+    </>
+  )
+}
+
+function Loading({ width = 50 }: { width?: number }) {
+  return <LoadingRow data-testid="loading-row" height={15} width={width} />
+}
+
+function ExchangeRateRow({ trade }: { trade: InterfaceTrade }) {
+  const { formatNumber } = useFormatter()
+  const rate = `1 ${trade.executionPrice.quoteCurrency.symbol} = ${formatNumber({
+    input: parseFloat(trade.executionPrice.toFixed(9)),
+    type: NumberType.TokenTx,
+  })} ${trade.executionPrice.baseCurrency.symbol}`
+  return <>{rate}</>
```

Comment:
dont think this fragment wrapper is necessary here

Reply:
() => string isn't a valid JSX ):

Reply:
RIP it should be, somethings up with our tsconfig